### PR TITLE
dev-lang/lua: Fixed pkgconfig configuration for slotted lua

### PR DESCRIPTION
**Pull request by winterheart**
Now generated lua.pc file produce correct includedir output
(/usr/include/lua5.[12]), fixes (really) #539828.

Fix for incorrect placement LMOD libraries. Previous ebuild had
left $V variable empty, so all libraries for lua placed to /usr/lib64/lua
instead /usr/lib64/lua/${SLOT}, fixes #524460.

There also warious fixes to tidy up and making repoman happy.

Gentoo-Bug: https://bugs.gentoo.org/524460
Gentoo-Bug: https://bugs.gentoo.org/539828

Package-Manager: portage-2.2.28

### CODE REVIEW

**Comment by winterheart** [comment]
@williamh

**Comment by Polynomial-C** [comment]
What's the status here? Why is this PR rotting for so long?

**Comment by williamh** [comment]
I have been pretty distracted by work stuff, but I'll look at it this weekend.

**Comment by mgorny** [comment]
@winterheart, GitHub says it no longer applies cleanly. Could you reevaluate this and update the PR, please?

**Comment by winterheart** [comment]
@mgorny done.

File: dev-lang/lua/lua-5.1.5-r100.ebuild [hunk 1]
```diff
@@ -107,8 +103,17 @@ multilib_src_install() {
...
-	insinto /usr/$(get_libdir)/pkgconfig
-	newins etc/lua.pc lua${SLOT}.pc
+	# We want packages to find our things...
+	sed \
+		-e "s:^prefix= :prefix= ${EPREFIX}:" \
+		-e "s:^V=.*:V= ${SLOT}:" \
+		-e "s:^R=.*:R= ${PV}:" \
+		-e "s:/,lib,:/$(get_libdir):g" \
+		-e "s:/,include,:/include/lua${SLOT}:g" \
+		"${FILESDIR}/lua.pc" > "${WORKDIR}/lua-$(get_libdir).pc"
```
Comment (mgorny) [inline dev-lang/lua/lua-5.1.5-r100.ebuild]: ||die

**Comment by winterheart** [comment]
Current slotted lua ebuilds are totally broken. Lua related packages cannot even compiles with them - they can't find -llua5.{1,2}. lua.pc returns -llua and empty cflags as described in https://bugs.gentoo.org/show_bug.cgi?id=539828 (I don't know why it marked as duplicated). 

Another huge issue is wrong INSTALL_LMOD (/usr/share/lua instead /usr/share/lua/5.{1,2}) and INSTALL_CMOD (/usr/lib/lua instead /usr/lib/lua5.{1,2}) in lua.pc. Lua-packages installs modules into wrong place.

Only enhancement here is dropping versionator.eclass since it not really required.

I agree that should be revbump or even verbump (5.2.4 is available).

**Comment by mgorny** [comment]
Ok, I see. Lemme ask a different question: I recall our Lua slotting was against upstream and practically forced patching every single package to support it. How do things look like these days? Are you able to build revdeps against slotted Lua? Do you need patches for that? Does it require you to set specific Lua slot via eselect?

**Comment by winterheart** [comment]
I believe slotting came to resolve building problems with packages that back and forward incompatibe with 5.1 / 5.2. Upstream says you should always bundle lua in this case, so providing shared libraries against upstream. Lua is trouble package, there is a lot of problems and lots of work for lua related packages. Currently none of them can compiles against slotted package. 

But I believe that slotting is right direction. Other distributives provides 5.1, 5.2 and 5.3 versions, so why we don't? Only one thing is there should be lua.eclass with LUA_TARGETS to ease and unify migration.

**Comment by winterheart** [comment]
@mgorny done.

**Comment by williamh** [comment]
I thought I would be around more this Thanksgiving but that's not how it worke out. I will catch up with rafaelmartins and see what he thinks about this. I can tell you that the lua shared library is probably going to go away since upstream doesn't support it as a shared library, but I haven't looked into the slotted lua setup yet.

**Comment by winterheart** [comment]
Why, shared libraries is less evil approach, since we shouldn't bundle lua into executables. Lua authors just don't see usefulness of shared libs in lua.
I'm thinking on using tarballs from LuaDist project (https://github.com/LuaDist/lua) since they done lot of work to provide sane build-system (based on CMake) with nice things such as ncurses and readline support. With these tarballs lua ebuilds greatly simplified.

**Comment by mgorny** [comment]
@williamh, I agree with @winterheart here. It's not shared libraries being the problem but using system-wide installs of Lua. Using static libs will only partially hide it — and you could get the same result (without drawbacks of static libs) using the slotting approach.

**Comment by monsieurp** [comment]
hi @winterheart. could you please rebase this PR? thanks.

**Comment by williamh** [comment]
Just a quick update on this. I have spoken with the other lua maintainer, and lua will be unslotted in the future due to multiple qa issues which upstream does not support.

**Comment by winterheart** [comment]
@monsieurp done.
@williamh why, why this decision was made? Upstream not aware of problems with our QA policies. They simply don't care. Maybe we should discuss this in the gentoo-dev?

**Comment by monsieurp** [comment]
hi. is this ready for prime time? @williamh @winterheart?

**Comment by williamh** [comment]
I have spoken with the other lua maintainer, @rafaelmartins, about this.

He, and flameeyes, who is our former qa lead, were never in favor of the lua eselect module  or slotting lua.

Also, upstream does not and will not support offering liblua as a shared library; this is mentioned in the lua ebuilds.


So, I'm against any more maintenance of slotted lua since it is going away. However, I have tagged @rafaelmartins, so we'll see if he feels differently.

**Comment by monsieurp** [comment]
OK. Let's close this PR then. Feel free to reopen it as you see fit.

**Pull request closed without merging**